### PR TITLE
API nested routes for host

### PR DESCRIPTION
**Pull request by isratrade**
Added nested routes
host/:id/reports
host/:id/reports/last
host/:id/audits
host/:id/puppetclasses
host/:id/facts

There is no route yet for host/:id/lookup_keys, since there is no host relation in lookup_key.rb to search_for

### CODE REVIEW

File: app/controllers/api/v1/audits_controller.rb [hunk 1]
```diff
@@ -10,7 +11,7 @@ class AuditsController < V1::BaseController
       param :per_page, String, :desc => "number of entries per request"
 
       def index
-        @audits = Audit.search_for(*search_options).paginate(paginate_options)
+        Audit.unscoped { @audits = Audit.search_for(*search_options).paginate(paginate_options) }
```
Comment (ohadlevy) [inline app/controllers/api/v1/audits_controller.rb:14]: why?

Comment (isratrade) [inline app/controllers/api/v1/audits_controller.rb:14]: @ohadlevy, I just copied from the UI audits_controller to the API audits_controller

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -56,5 +56,8 @@
         <%= link_to 'Support', "http://theforeman.org/projects/foreman/wiki/Support", :rel => "external" %>
       </span>
     </div>
+    <% if Rails.env.development? %>
+        <%= debug(params) %>
```
Comment (ohadlevy) [inline app/views/layouts/application.html.erb:60]: debug left over?

Comment (isratrade) [inline app/views/layouts/application.html.erb:60]: @ohadlevy, you don't need to include this.

**Comment by ohadlevy** [comment]
@isratrade please cleanup and rebase, thanks

**Comment by isratrade** [comment]
Closed and re-submitted / re-based in pull request #277

**Pull request closed without merging**